draft-ietf-tls-esni.md

```diff
@@ -723,38 +716,56 @@ error code.
 
 ### Handling HelloRetryRequest {#client-hrr}
 
-As required in {{real-ech}}, clients offering ECH MUST ensure that all
-extensions or parameters that might change in response to receiving a
-HelloRetryRequest have the same values in ClientHelloInner and
-ClientHelloOuter. That is, if a HelloRetryRequest causes a parameter to be
-changed, the same change is applied to both ClientHelloInner and
-ClientHelloOuter. Applicable parameters include:
+If the server sends a HelloRetryRequest in response to the ClientHello, the
+client first checks for the presence of the "encrypted_client_hello" extension.
+The presence of this extension is an indication that the client-facing server
+accepted ECH and that the backend server sent a HelloRetryRequest. If absent,
+then the client presumes rejection and proceeds as if the message was sent by
+the client-facing server.
+
+The HelloRetryRequest constructed by the backend server is called the
+HelloRetryRequestInner message; and the HelloRetryRequest constructed by the
+client-facing server is called the HelloRetryRequestOuter message.
+
+If the client offered ECH in the first ClientHello, then it MUST offer ECH in
+the second. Likewise, if the client did not offer ECH in the first ClientHello,
+then it MUST NOT not offer ECH in the second.
```

Comment:
This also is unclear without the text ahead. I would either summarize the behavior or have a forward reference.

Reply:
You know, this paragraph doesn't seem relevant anymore, since the text now prescribes exactly how CHInner and CHOuter are constructed after HRR. I just deleted it.